user_signup/main.py

```diff
@@ -0,0 +1,86 @@
+import webapp2
+import jinja2
+import os
+import re
+
+jinja_env = jinja2.Environment(
+	loader=jinja2.FileSystemLoader(os.path.dirname(__file__)), 
+	autoescape=True)
+
+USER_RE = re.compile(r"^[a-zA-Z0-9_-]{3,20}$")
+PW_RE = re.compile(r"^.{3,20}$")
+EMAIL_RE = re.compile(r"^[\S]+@[\S]+.[\S]+$")
```

Comment:
Did you write these regular expressions?  Does this regex correctly validate all possible emails?